src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs

```diff
@@ -106,7 +110,7 @@ class NoUnexpectedThunks a where
   whnfNoUnexpectedThunks :: [String] -> a -> IO ThunkInfo
   default whnfNoUnexpectedThunks :: (Generic a, GWhnfNoUnexpectedThunks (Rep a))
                                  => [String] -> a -> IO ThunkInfo
-  whnfNoUnexpectedThunks ctxt x = gWhnfNoUnexpectedThunks ctxt fp
+  whnfNoUnexpectedThunks ctxt x = gWhnfNoUnexpectedThunks [] ctxt fp
```

Comment:
I was surprised to see you pass this list at the _value level_, rather than the type level; all metadata is available at the type level, and so all decisions could be made at the type level. But I guess this way works also. Feels slightly less clean, but no big deal.

Reply:
Do you have an approach in mind for doing it at the type level?

I'd be up for it.

Reply:
Ah, I think I see it. Let's see how it compares.